### PR TITLE
Add CVS/JSON export button

### PR DESCRIPTION
**Pull request by rvsia**
https://issues.redhat.com/browse/RHCLOUD-14766

**After**

Kapture 2021-06-30 at 09 17 41

TODO:
- [x] tests

### CODE REVIEW

**Review by Hyperkid123 — APPROVED** [review]

File: src/pages/Sources.js [hunk 1]
```diff
@@ -264,6 +269,16 @@ const SourcesPage = () => {
...
           onDelete: (_event, chips, deleteAll) => dispatch(filterSources(removeChips(chips, filterValue, deleteAll))),
         }}
+        exportConfig={{
+          isDisabled: !loaded,
+          onSelect: (_e, type) => {
+            const data =
+              type === 'csv'
+                ? generateCSV(entities, intl, appTypes, sourceTypes)
+                : generateJSON(entities, intl, appTypes, sourceTypes);
+            downloadFile(data, `sources-${new Date().toISOString()}`, type);
```
Comment (Hyperkid123) [inline src/pages/Sources.js:279]: As this is very dependent on the API response shape and can cause runtime errors, can we add a try-catch block around the action? And possibly display error notifications if the CSV/JSON conversion fails?

Comment (rvsia) [reply src/pages/Sources.js:279]: If the response is different than the whole app would fail. 🤷

Comment (Hyperkid123) [reply src/pages/Sources.js:279]: Spicy :hot_pepper:

Comment (rvsia) [reply src/pages/Sources.js:279]: There is the error boundary, so it should be ok. 😏 

But the whole application is based on this one API response, so if it were changed, then the whole application would be changed and there are integration tests that would catch all the differences.

**Pull request merged**